src/CatSearch.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from "react"
+import { useState, useEffect, useCallback } from "react"
+import { fetchSearchResultsFromAPI } from "./searchApiClient"
+
+export function CatSearch() {
+  const [valid, setValid] = useState(false)
+  const [hits, setHits] = useState<any[]>([])
+  const [searchTerm, setSearchTerm] = useState("")
+
+  const searchTermOnChange = useCallback((event) => {
+    const { value } = event.target
+    setSearchTerm(value)
+  }, [])
+
+  useEffect(() => {
+    setValid(true)
```

Comment:
After we add the dependency array for `searchTerm`, this line will introduce a bug where we will set `valid` to true for a split second when we have a single character input.

To fix this, we can simple move this line after our validation check.